src/app/components/data-picker/category-picker/terra-category-picker.component.ts

```diff
@@ -203,10 +209,29 @@ export class TerraCategoryPickerComponent extends TerraNestedDataPickerComponent
                     {
                         categoryDetail = categoryData.details.find((foundDetail:CategoryDetailDataInterface) =>
                         {
-                            return foundDetail.lang === this.inputLanguage;
+                            return foundDetail.lang === this.inputLanguage && (foundDetail.plentyId === this.inputPlentyId ||
+                                                                               isNullOrUndefined(this.inputPlentyId));
                         });
 
-                        // No details found with the given language so just use the first language instead
+                        // Check if there is a detail only for the language
+                        if(isNullOrUndefined(categoryDetail))
+                        {
+                            categoryDetail = categoryData.details.find((foundDetail:CategoryDetailDataInterface) =>
+                            {
+                                return foundDetail.lang === this.inputLanguage;
+                            });
+                        }
+
+                        // Check if there is a detail only for the plentyId
+                        if(isNullOrUndefined(categoryDetail))
+                        {
+                            categoryDetail = categoryData.details.find((foundDetail:CategoryDetailDataInterface) =>
+                            {
+                                return foundDetail.plentyId === this.inputPlentyId || isNullOrUndefined(this.inputPlentyId);
+                            });
+                        }
+
+                        // No details found with the given language and the given plentyId so just use the first language instead
                         if(isNullOrUndefined(categoryDetail))
                         {
                             categoryDetail = categoryData.details[0];
```

Comment:
hier auch bitte noch eine Abfrage mit rein ob categoryData.details.length > 0 ist. Es gab in manchen System categories wo kein einziges detail einer category zugeordnet war